v1/src/test/java/com/google/cloud/teleport/templates/TextToPubsubIT.java

```diff
@@ -81,24 +81,27 @@ public void testTextToTopic() throws IOException {
     // Act
     JobInfo info = launchTemplate(options);
     assertThat(info.state()).isIn(JobState.ACTIVE_STATES);
-    AtomicReference<PullResponse> records = new AtomicReference<>();
+    List<String> records = new ArrayList<>();
     Result result =
         new DataflowOperator(getDataflowClient())
-            .waitForConditionAndFinish(
+            .waitForCondition(
```

Comment:
If I got it correctly, you are not cancelling / finishing it here because it's a batch job?